### PR TITLE
Removed multiple iterations of corpus in p_boolean_document.

### PR DESCRIPTION
**Pull request by danielchamberlain**
In the previous version of p_boolean_document, the entire
corpus was iterated through for each top_id. In this new
version, the corpus is iterated through a single time
and the matching top_ids for each document are extracted
simultaneously.

### CODE REVIEW

**Comment by danielchamberlain** [comment]
The build failed because of trailing whitespace after a colon. Should I submit a new pull request?

**Comment by menshikh-iv** [comment]
@danielchamberlain No, you can fix it in current PR

**Comment by danielchamberlain** [comment]
Thanks. All set now.

**Comment by tmylk** [comment]
Thanks for the improvement. Confirm regression test `testPBooleanDocument` passe.

File: gensim/topic_coherence/probability_estimation.py [hunk 1]
```diff
@@ -47,14 +47,17 @@ def p_boolean_document(corpus, segmented_topics):
...
-        for n, document in enumerate(corpus):
-            if id in frozenset(x[0] for x in document):
-                id_list.add(n)
-        per_topic_postings[id] = id_list
+        per_topic_postings[id] = set()
+    # Iterate through the documents, appending the document number to the set for each top_id it contains
+    for n, document in enumerate(corpus):
+        doc_words = frozenset(x[0] for x in document)
+        top_ids_in_doc = top_ids.intersection(doc_words)
+        if len(top_ids_in_doc) > 0:
```
Comment (piskvorky) [inline gensim/topic_coherence/probability_estimation.py]: `if top_ids_in doc:` in Python.

File: gensim/topic_coherence/probability_estimation.py [hunk 2]
```diff
@@ -47,14 +47,17 @@ def p_boolean_document(corpus, segmented_topics):
...
+            for id in top_ids_in_doc:
+                per_topic_postings[id].add(n)
```
Comment (piskvorky) [inline gensim/topic_coherence/probability_estimation.py]: Isn't `per_topic_postings.update(top_ids.intersection(doc_words))` both simpler, easier to read and faster?

**Pull request merged**